frame/bridge/ethereum/issuing/src/lib.rs

```diff
@@ -0,0 +1,429 @@
+	type EcdsaAuthorities: RelayAuthorityProtocol<Self::BlockNumber, Signer = EthereumAddress>;
+	type WeightInfo: WeightInfo;
+	type FeeEstimate: Get<RingBalance<Self>>;
+}
+
+decl_error! {
+	/// Issuing pallet errors.
+	pub enum Error for Module<T: Config> {
+		/// Invalid Issuing System Account
+		InvalidIssuingAccount,
+		/// assert ar
+		AssetAR,
+		/// ReceiptProofInv
+		ReceiptProofInv,
+		/// LogEntryNE
+		LogEntryNE,
+		/// EthLogPF
+		EthLogPF,
+		/// StringCF
+		StringCF,
+		/// Unit
+		UintCF,
+		/// Address - CONVERSION FAILED
+		AddressCF,
+		/// encode erc20 tx failed
+		InvalidEncodeERC20,
+		/// encode mint tx failed
+		InvalidMintEcoding,
+		/// invalid ethereum address length
+		InvalidAddressLen,
+		/// decode input value error
+		InvalidInputData,
+	}
+}
+
+decl_event! {
+	pub enum Event<T>
+	where
+		AccountId = AccountId<T>,
+	{
+        /// redeem transaction from ethereum
+		RegisteredOrRedeemed(AccountId, EthereumAddress),
+		/// erc20 created
+		CreateErc20(EthereumAddress),
+		/// burn event
+		/// type: 1, backing, recipient, delegator, source, target, value
+		BurnToken(u8, EthereumAddress, EthereumAddress, EthereumAddress, EthereumAddress, EthereumAddress, U256),
+		/// token registed event
+		/// type: u8 = 0, backing, source(origin erc20), target(mapped erc20)
+		TokenRegisted(u8, EthereumAddress, EthereumAddress, EthereumAddress),
+	}
+}
+
+decl_storage! {
+	trait Store for Module<T: Config> as DarwiniaEthereumIssuing {
+		pub MappingFactoryAddress get(fn mapping_factory_address) config(): EthereumAddress;
+		pub VerifiedIssuingProof
+			get(fn verified_issuing_proof)
+			: map hasher(blake2_128_concat) EthereumTransactionIndex => bool = false;
+		pub BurnTokenEvents get(fn burn_token_events): Vec<<T as frame_system::Config>::Event>;
+	}
+}
+
+decl_module! {
+	pub struct Module<T: Config> for enum Call
+	where
+		origin: T::Origin
+	{
+		const FeeEstimate: RingBalance<T> = T::FeeEstimate::get();
+
+		fn deposit_event() = default;
+
+		fn on_initialize(_n: BlockNumber<T>) -> Weight {
+			<BurnTokenEvents<T>>::kill();
+			0
+		}
+
+		#[weight = <T as darwinia_evm::Config>::GasWeightMapping::gas_to_weight(0x100000)]
+		pub fn register_or_redeem_erc20(origin, backing: EthereumAddress, proof: EthereumReceiptProofThing<T>) {
+			log::trace!(target: "darwinia-issuing", "start to register_or_issuing_erc20");
+			let user = ensure_signed(origin)?;
+			let tx_index = T::EthereumRelay::gen_receipt_index(&proof);
+			ensure!(!VerifiedIssuingProof::contains_key(tx_index), <Error<T>>::AssetAR);
+			let verified_receipt = T::EthereumRelay::verify_receipt(&proof)
+				.map_err(|err| {
+					log::trace!(target: "darwinia-issuing", "verify error {:?}", err);
+					<Error<T>>::ReceiptProofInv
+				})?;
+
+			let register_event = Abi::register_event();
+			let backing_event = Abi::backing_event();
+			let log_entry = verified_receipt
+				.logs
+				.into_iter()
+				.find(|x| {
+					x.address == backing &&
```

Comment:
Anyone can call this dispatch call, and the backing is passed by the user, so he can specify any backing address as he want.

So it is useless for this validation, and backing can be removed from call parameters.

And as discussed, to prevent too many backing and mapping tokens spam in the bridge, we will only use one backing for each source chain.

Reply:
how can we change this parameter as network Id?  and then all other ethereum-like chains can be processed by a common way.

Reply:
Yes, we need a network id in the event and proof. Since we will verify the proof and extract that value from the proof.

Passing the network id as call parameter is meaningless for the same reason.

Reply:
I need to correct, if consider the network id, we may need that in the call parameter since we have to know which network's relay will be used to verify this receipt, only after that, we can tell the receipt format and info.

But for here in ethereum, we do not need network_id in the event/receiot.